### ISSUE TITLE
Question: "anyone with the link can view" permission

### ISSUE THREAD

**Issue by RickCogley** [odeke-em/drive#568]
Hi - how could I set "anyone with the link can view" permission like in the inline screenshot below? The ``drive share`` command seems to be email-centric. Please advise, thanks. 

![jrc screenshot 2016-01-26 at 11 29 14 pm](https://cloud.githubusercontent.com/assets/512328/12583208/e94a7c60-c484-11e5-9905-9c39b96901b7.png)

**Comment by clawoflight** [comment]
Use `drive pub`.

**Comment by RickCogley** [comment]
Hi @clawoflight, thanks. ``drive pub`` makes the link public, which google says is indexed. The above setting is apparently different from that, in that it is not indexed.

**Comment by odeke-em** [comment]
Hello again @RickCogley. Unfortunately `drive pub` is the only way right now. I didn't know about this nor how to solve this currently. Any doc references?

**Comment by RickCogley** [comment]
I looked at the API docs, but, I cannot see anything specific about those two settings which are available in the GUI itself. @odeke-em

**Comment by thebaddie** [comment]
Here's what's needed: https://developers.google.com/drive/v3/reference/permissions#resource-representations

type: "anyone"
role: "reader"
allowFileDiscovery: false

It would be very usefull to add the "anyone with the link" feature

**Comment by odeke-em** [comment]
Aha I finally now understand the problem. Thank you @thebaddie for rephrasing it and thank you @RickCogley for advocating it, I have been able to locally produce the expected output. PR coming shortly.

**Comment by odeke-em** [comment]
Hey folks. Thank you very much for the patience. I finally got time and thought about it. After re-reading the content in the issue I was able to discover how to solve it. The hints and discussions helped a lot. I learnt something about published files being indexed (which is not want the user wants) and also thanks for the reference to the documentation(we were using the same permissions anyways, but a refresher was good). Auto-closed by PR #648. Please get the latest from master
```shell
$ go get -v -u github.com/odeke-em/drive/drive-gen && drive-gen
```
and then you can now do this
```shell
$ drive share --type user,anyone,anyone,user --role reader,anyone --emails "emm.odeke@gmail.com" outf.gif
Provide access for accountType(s)
  user
  anyone

For roles(s)
  reader

Addressees:
  + emm.odeke@gmail.com
  + Anyone with the link

File(s) to share:
  + outf.gif

Proceed with the changes? [Y/n]:y
successful share for outf.gif with email "emm.odeke@gmail.com", role "reader" accountType "user"
successful share for outf.gif with email "", role "reader" accountType "anyone"
```
Then get the URL of the file
```shell
$ drive url outf.gif
```

Please reopen if it persists.

**Comment by thebaddie** [comment]
Testing it right now but it still is public and indexed, 
maybe is the allowFileDiscovery: false missing?

![Sharing screenshot](http://img.prntscr.com/img?url=http://i.imgur.com/IEd5JE1.png)
